frontend/src/api/Socket.ts

```diff
@@ -1,54 +1,92 @@
+/* eslint-disable consistent-return */
 import SockJS from 'sockjs-client';
 import Stomp, { Client, Message } from 'stompjs';
 
 let stompClient: Client;
 
-type User = {
+export type User = {
   nickname: string;
 }
 
 // Variable to hold the current connected state.
-let connected = false;
+let connected: boolean = false;
 
 // Create constants for the connection, subscription, and send message URLs.
 export const SOCKET_ENDPOINT:string = '/api/v1/socket/join-room-endpoint';
 const SUBSCRIBE_URL:string = '/api/v1/socket/subscribe-user';
-const SEND_USER_URL:string = '/api/v1/socket/user';
+const ADD_USER_URL:string = '/api/v1/socket/add-user';
+const DELETE_USER_URL:string = '/api/v1/socket/delete-user';
 
-export const sendUser = (nickname:string) => {
-  if (connected) {
-    stompClient.send(SEND_USER_URL, {}, nickname || 'Anonymous');
+export const isValidNickname = (nickname: string) => nickname.length > 0
+  && !nickname.includes(' ') && nickname.length <= 16;
+
+/**
+ * Add the user by sending a message via socket.
+ * @returns error, if present
+*/
+export const addUser = (nickname:string): Error | undefined => {
+  if (connected && isValidNickname(nickname)) {
+    stompClient.send(ADD_USER_URL, {}, nickname);
+  } else if (!connected) {
+    return new Error('The socket is not connected.');
   } else {
-    console.error('You must be connected to a socket before sending user information.');
+    return new Error('The provided nickname is invalid.');
   }
 };
 
-export const connect = (endpoint:string, nickname:string) => {
-  if (!connected) {
-    // Connect to given endpoint, subscribe to future messages, and send user message.
-    const socket: WebSocket = new SockJS(endpoint);
-    stompClient = Stomp.over(socket);
-    stompClient.connect({}, () => {
-      stompClient.subscribe(SUBSCRIBE_URL, (user: Message) => {
-        const userObject:User = JSON.parse(user.body);
-        console.log(`Welcome ${userObject.nickname} to the page!`);
-      });
-      sendUser(nickname);
-    });
-    // Reassign connected variable.
-    connected = true;
+/**
+ * Delete the user by sending a message via socket.
+ * @returns error, if present
+*/
+export const deleteUser = (nickname:string): Error | undefined => {
+  if (connected && isValidNickname(nickname)) {
+    stompClient.send(DELETE_USER_URL, {}, nickname);
+  } else if (!connected) {
+    return new Error('The socket is not connected.');
   } else {
-    console.error('You are already connected to a socket.');
+    return new Error('The provided nickname is invalid.');
   }
 };
 
-export const disconnect = () => {
+/**
+ * Connect and subscribe the user via socket.
+ * @returns error, if present
+*/
+export const connect = (endpoint:string, nickname:string,
+  setUsers: React.Dispatch<React.SetStateAction<User[]>>):
+  Promise<Error | undefined> => new Promise<Error | undefined>((resolve, reject) => {
```

Comment:
For consistency with how you do it in the above functions, are you able to just resolve the Error object instead of rejecting? That way we don't have to deal with catch blocks in our React files, and then you might also be able to simplify this line to `Promise<Error | undefined>`

Reply:
Hmm ok, I'm sorry I didn't realize that before; I'd want to discuss maybe using `catch` blocks instead? Wouldn't it be smarter to use that built-in block rather than calling `isError(...)`?

I'll also note that I used `catch(...)` in a branch on top of this as well -- not that it matters, but just I think it might be smart to switch to that? (See [catch block from next PR](https://github.com/rocketden/main/blob/ed5b3253a252e58548e0bcfb8c9ed50c55feedbc/frontend/src/views/Join.tsx#L48).)

Reply:
After our conversation, I decided to switch the codebase to use `catch(...)`. I am updating the old functions as well right now.

Reply:
The old functions are updated!